src/Raven.Server/Documents/Patch/PatchDocumentCommand.cs

```diff
@@ -233,6 +235,73 @@ BlittableObjectInstance UpdateOriginalDocument()
             }
         }
 
+        private void AssertNoModificationsToAttachments(BlittableJsonReaderObject originalDoc, BlittableJsonReaderObject modifiedDocument, string id)
+        {
+            originalDoc.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject originalMetadata);
+            modifiedDocument.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject modifiedMetadata);
+            if (originalMetadata == null && modifiedMetadata == null)
+                return;
+            BlittableJsonReaderArray originalAttachments = null;
+            BlittableJsonReaderArray modifiedAttachments = null;
+            originalMetadata?.TryGet(Constants.Documents.Metadata.Attachments, out originalAttachments);
+            modifiedMetadata?.TryGet(Constants.Documents.Metadata.Attachments, out modifiedAttachments);
+            if(originalAttachments == null && modifiedAttachments == null)
+                return;
+
+            if (originalAttachments == null || modifiedAttachments == null)
+            {
+                ThrowPatchModificationToAttachments(id);
+            }
+
+            if (originalAttachments.Length != modifiedAttachments.Length)
+            {
+                ThrowPatchModificationToAttachments(id);
+            }
+
+            for (int i = 0; i< originalAttachments.Length; i++)
+            {
+                var attachmentsForOriginalDocument = originalAttachments[i] as BlittableJsonReaderObject;
+                var attachmentsForModifiedDocument = modifiedAttachments[i] as BlittableJsonReaderObject;
+
+                if(attachmentsForOriginalDocument == null || attachmentsForModifiedDocument == null)
+                {
+                    ThrowPatchModificationToAttachments(id);
+                }
+
+                if (attachmentsForOriginalDocument.Count != attachmentsForModifiedDocument.Count)
+                {
+                    ThrowPatchModificationToAttachments(id);
+                }
+
+                unsafe
+                {
+                    if(Sparrow.Memory.Compare(attachmentsForOriginalDocument.BasePointer, attachmentsForModifiedDocument.BasePointer, attachmentsForOriginalDocument.Count) != 0)
+                    {
+                        ThrowPatchModificationToAttachments(id);
+                    }
+                }
+
+            }
+        }
+
+        private static Dictionary<string, BlittableJsonReaderObject> GetAttachmentsFromArray(BlittableJsonReaderArray attachmentsArray)
```

Comment:
I don't think that this method is used anywhere?

Reply:
seems i forgot to clean this one up